crates/core/tedge_agent/src/tedge_operation_converter/actor.rs

```diff
@@ -119,37 +124,35 @@ impl TedgeOperationConverterActor {
     }
 
     async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), RuntimeError> {
-        let (operation, cmd_id) = match self.mqtt_schema.entity_channel_of(&message.topic) {
-            Ok((_, Channel::Command { operation, cmd_id })) => (operation, cmd_id),
+        let Ok((_, operation, cmd_id)) = self.extract_command_identifiers(&message.topic.name)
+        else {
+            log::error!("Unknown command channel: {}", &message.topic.name);
+            return Ok(());
+        };
 
-            _ => {
-                log::error!("Unknown command channel: {}", message.topic.name);
-                return Ok(());
-            }
+        let Ok(state) = GenericCommandState::from_command_message(&message) else {
+            log::error!("Invalid command payload: {}", &message.topic.name);
+            return Ok(());
         };
+        let step = state.status.clone();
 
-        let mut log_file = CommandLog::new(self.log_dir.clone(), &operation, &cmd_id).await;
-        match self.workflows.apply_external_update(&operation, &message) {
-            Ok(None) => {
-                if message.payload_bytes().is_empty() {
-                    log_file
-                        .log_step("", "The command has been fully processed")
-                        .await;
-                    self.persist_command_board().await?;
-                }
-            }
-            Ok(Some(state)) => {
+        let mut log_file =
+            self.open_command_log(state.invoking_command_topic(), &operation, &cmd_id);
```

Comment:
```suggestion
            self.open_command_log(state.root_command_topic(), &operation, &cmd_id);
```
Not that it'll work now, but that would be required for nested ops support.

Reply:
Oops! However the fix is more subtle as the root command topic cannot be inferred directly from the command state.

Reply:
fixed by https://github.com/thin-edge/thin-edge.io/pull/2792/commits/d6aef9a517a9112ae229509bfc1f118eb35dc97a